syntax = "proto3";

package PbConfigs;

message CanDevice
{
    string socket = 1;
    string name = 2;
    repeated string networks = 3;
}

message GpsDevice
{
    string address = 1;
    string mode = 2;
    bool enabled = 3;
}

message Connection
{
    string ip = 1;
    string port = 2;
    string mode = 3;
    string whoamiUrl = 4;
    bool tlsEnabled = 5;
    string cafile = 6;
    string capath = 7;
    string certfile = 8;
    string keyfile = 9;
}

message ConnectionSettings
{
    bool enabled = 1;
    bool downsampleEnabled = 2;
    bool downsampleSkipData = 3;
    int64 downsampleMps = 4;
    int64 sendRate = 5;
    bool sendSensorData = 6;
}

message TelemetryConfig
{
    bool cameraEnabled = 4;
    bool generateCsv = 5;
    Connection connection = 6;
    ConnectionSettings connectionSettings = 7;
    repeated CanDevice canDevices = 8;
    repeated GpsDevice gpsDevices = 9;
}